runner/pool/pool.go

```diff
@@ -246,6 +263,20 @@ func (r *basePoolManager) HandleWorkflowJob(job params.WorkflowJob) error {
 			return errors.Wrap(err, "converting job to params")
 		}
 
+		// Runner name was not set in WorkflowJob by github. We can still attempt to fetch the info we need,
+		// using the workflow run ID, from the API.
+		if job.WorkflowJob.RunnerName == "" {
+			// We may still get no runner name. In situations such as jobs being cancelled before a runner had the chance
+			// to pick up the job, the runner name is not available from the API.
+			if job.WorkflowJob.Conclusion != "skipped" && job.WorkflowJob.Conclusion != "canceled" {
```

Comment:
There is very little chance that a workflow job that is `in_progress` will have the conclusion set to `skipped` or `canceled`. I think we can drop this conditional statement. I am curious if the GH API would be so flaky as to not have the runner name set in a job that is marked as `in_progress`. I think that would be a huge bug, and we shouldn't really account for it.

Long story short, I think we can drop this check and just leave the one in the case where the job is `completed`. 

Maybe just error our if the name is empty.

Reply:
Was already wondering about this condition, tbh.
To get a better understanding, i would like to add few more debug-log messages in here to be able to see some of the relevant job events we get from GH.